pkg/services/requests.go

```diff
@@ -37,13 +43,34 @@ func GenerateRequestsTable(requests []clientapi.AccessRequestClientModel) *uitab
 			revocationTime = "NA"
 		}
 
-		table.AddRow(request.Id, utils.DisplayTime(creationTime), revocationTime, integrations, *request.Justification.Get(), coloredStatus(request.Status.Status))
+		table.AddRow(request.Id, utils.DisplayTime(creationTime), revocationTime, integrations, *request.Justification.Get(), ColoredStatus(request))
 	}
 
 	return table
 }
 
-func GetUserLastRequest(ctx context.Context, client *aponoapi.AponoClient) (*clientapi.AccessRequestClientModel, error) {
+func WaitForNewRequest(ctx context.Context, client *aponoapi.AponoClient, creationTime time.Time, timeout time.Duration) (*clientapi.AccessRequestClientModel, error) {
+	startTime := time.Now()
+	for {
+		lastRequest, err := getUserLastRequest(ctx, client)
+		if err != nil {
+			return nil, err
+		}
+
+		lastRequestCreationTime := utils.ConvertUnixTimeToTime(lastRequest.CreationTime)
+		if lastRequestCreationTime.After(creationTime) {
+			return lastRequest, nil
+		}
+
+		time.Sleep(1 * time.Second)
+
+		if time.Now().After(startTime.Add(timeout)) {
+			return nil, fmt.Errorf("timeout while waiting for request to be created")
+		}
+	}
+}
```

Comment:
moved from `pkg/commands/requests/actions/new.go` to here, and added timeout argument.